voc/python/klass.py

```diff
@@ -51,6 +58,10 @@ def __init__(
         self._constructor = None
 
     @property
+    def full_name(self):
+        return '.'.join(self.namespace.split('.')[1:] + [self.name])
```

Comment:
This is more involved than I expected… Some thoughts while I wait for the tests to finish. (Also see: commit message of a75bc9c.)

Say we have this structure:

```python
class MyClass1:
    class InnerClass
        ...
```

The inner class would be called `__main__.MyClass1.InnerClass` (Voc currently names it `MyClass1.InnerClass`, which is wrong, but that problem is for another day), but in JVM its descriptor needs to be `python/MyClass$InnerClass`. Previously a class’s descriptor is calculated by replacing `.`s with `/`s, but with inner classes in the picture we need to know when to use `$` or `/`.

I use a different logic behind `class_name`, `full_name`, and `descriptor`. For the former two, a class can get its own value by getting its parent’s value, and simply append `.{self.name}`; for the descriptor, however, only its parent knows whether it needs `$` or `/` before its name. I delegated the logic to `build_child_class_descriptor()` so that the parent can lend a hand on this.

Everything after is pretty straightforward. An inner class’s parent should be its outer class, and by implementing the same properties duck-typing takes care of the rest. The transpiled class name also needed changing.